pcdsdevices/lodcm.py

```diff
@@ -84,13 +158,129 @@ class LODCM(BaseInterface, Device):
 
     mono_line : str, optional
         Name of the mono, double-bounce beamline.
-    """
 
+    z1_prefix : str
+        LOM Xtal1 Z motor prefix.
+
+    x1_prefix : str
+        LOM Xtal1 X motor prefix.
+
+    y1_prefix : str
+        LOM Xtal1 Y motor prefix.
+
+    th1_prefix : str
+        LOM Xtal1 Theta motor prefix.
+
+    ch1_prefix : str
+        LOM Xtal1 Chi motor prefix.
+
+    h1n_m_prefix : str
+        LOM Xtal1 Hn motor prefix.
+
+    h1p_prefix : str
+        LOM Xtal1 Hp motor prefix.
+
+    z2_prefix : str
+        LOM Xtal2 Z motor prefix.
+
+    x2_prefix : str
+        LOM Xtal2 X motor prefix.
+
+    y2_prefix : str
+        LOM Xtal2 Y motor prefix.
+
+    th2_prefix : str
+        LOM Xtal2 Theta motor prefix.
+
+    ch2_prefix : str
+        LOM Xtal2 Chi motor prefix.
+
+    h2n_prefix : str
+        LOM Xtal2 Hn motor prefix.
+
+    diode2_prefix : str
+        LOM Xtal2 PIPS motor prefix.
+
+    dh_prefix : str
+        LOM Dia H motor prefix.
+
+    dv_prefix : str
+        LOM Dia V motor prefix.
+
+    dr_prefix : str
+        LOM Dia Theta motor prefix.
+
+    df_prefix : str
+        LOM Dia Filter Wheel motor prefix.
+
+    dd_prefix : str
+        LOM Dia PIPS motor prefix.
+
+    yag_zoom_prefix : str
+        LOM Zoom motor prefix.
+    """
     h1n = Cpt(H1N, ':H1N', kind='hinted')
     yag = Cpt(YagLom, ":DV", kind='omitted')
     dectris = Cpt(Dectris, ":DH", kind='omitted')
     diode = Cpt(Diode, ":DIODE", kind='omitted')
     foil = Cpt(Foil, ":FOIL", kind='omitted')
+    y1_state = Cpt(Y1, ':Y1', kind='omitted')
+    chi1_state = Cpt(CHI1, ':CHI1', kind='omitted')
+    h2n_state = Cpt(H2N, ':H2N', kind='omitted')
+    y2_state = Cpt(Y2, ':Y2', kind='omitted')
+    chi2_state = Cpt(CHI2, ':CHI2', kind='omitted')
+    # TODO energy component??
+    energy = Cpt(PseudoSingleInterface, egu='eV')
+
+    # Crystal Tower 1
```

Comment:
I'm wondering if it would make more sense to have a separate class for each Tower... might make it cleaner but because i'm not entirely sure how this class is supposed to behave yet I don't know if there will be something ahead that would complicate things that way

Reply:
Looking at @ZLLentz 's image with the descriptions, 3 tower classes seems appropriate - assuming their calculations are independent of one another. 

Instead of trying to figure that out by studying the code, asking someone who knows about it will save a bunch of time/effort here...

Reply:
I think making separate tower classes is appropriate if it helps organize the code. Unfortunately the towers don't share too many commonalities with each other and there will be some redundancy in the naming, like:
lodcm.tower1.z1
lodcm.tower2.z2
but there wouldn't be a z1 on tower2

Reply:
Now that you have spoken with Dilingm you are likely aware of the desire of the scientists to type as few characters as possible, so we will need to make aliases if we add another component that would require an extra tab or character to type in a 'user facing' device. They will want lom_z1  as this is what they had (lodcm_z1 could also work as they should be able to tab complete after lod

Reply:
Yes that's correct Silke, I think if I end up making separated classes I would still define all those motors in LODCM class, something I did for Kappa for the XYZStage:

```
    self.z1 = self.first_tower.z1
    self.x1 = self.first_tower.x1
    ...... and so on....
```
but I need to make sure they are available when using `tab` though, not sure if they will be this way....